Added options for using git data repo

scuba75 opened the pull request.

scuba75 committed:
Added options for using git data repo

scuba75 committed:
corrected data update using git repo

scuba75 committed:
Merge branch 'main' into 9-add-option-to-pull-game-data-collections-from-github-repo

Lego-Fan9 commented:
Wouldn’t it be better to pull the metadata from the GitHub repo as well? Since it updates at the same time as the data updates. Would also make it non-reliant on comlink I think.

MarTrepodi commented:
> Wouldn’t it be better to pull the metadata from the GitHub repo as well? Since it updates at the same time as the data updates. Would also make it non-reliant on comlink I think.

I think it is best to keep the metadata pull via Comlink since it is light weight in comparison and will give the most up to date information possible. As @scuba75 mentioned, there are times when the gamedata repo is not updated as reliably as it could be.

scuba75 committed:
fix git data downloader

scuba75 committed:
Add Brotli decompression and error handling

Updated the gitDataDownloader.js file to include decompression of Brotli compressed files and improved error handling for fetching data.

scuba75 committed:
Refactor gitDataDownloader.js for improved structure

scuba75 committed:
fix git download
